spec/requests/application_insights_spec.rb

```diff
@@ -0,0 +1,27 @@
+require "rails_helper"
+
+RSpec.describe "Application Insights", type: :request do
+  let(:app) { lambda { |env| [200, {"Content-Type" => "text/plain"}, ["OK"]] } }
+  let(:request) { Rack::MockRequest.new(subject) }
+  let(:send_interval) { 0.2 }
+  let(:ip) { "1.1.1.1" }
+
+  subject { ApplicationInsights::Rack::TrackRequest.new(app, "ABCDEFG", send_interval) }
+
+  it "sends the client IP along with the rest of the data" do
+    # Set up a stub for a request to Application Insights that includes the
+    # client IP. The request is gzipped, so requires us to expand the response
+    # and then check the contents of the json
+    stub = stub_request(:post, ApplicationInsights::Channel::AsynchronousSender::SERVICE_ENDPOINT_URI).with { |req|
+      body = Zlib::GzipReader.new(StringIO.new(req.body)).readlines
+      json = JSON.parse(body[0])
+      json[0]["data"]["baseData"]["properties"]["clientIp"] == ip
+    }
+    request.get("/", {"REMOTE_ADDR" => ip})
+    # The Application Insights resquests are sent asynchronously, so we need to
+    # sleep for the value of `send_interval` (which is how often the logs are sent)
+    # before we check the stub has been requested
+    sleep(send_interval)
```

Comment:
I've tried setting it to `0` and the test fails. My guess is it's a race condition because the request it happening in a different thread, so the two things are happening at exactly the same time, so that small sleep allows for the two threads to get back into sync.

Reply:
Does it make sense that it's exactly the same as the `send_interval`? I would expect it to be longer (like 1.5x), to be more reliable about the interval having passed and the other thread having finished executing.

Reply:
It's stubbed, so should be pretty much instantaneous

Reply:
Can we make that clearer in the spec, then? Like it doesn't make sense to share the value between them, does it?